File: chat2db-community-server/chat2db-community-query-excel/query-excel-domain-api/src/main/java/ai/chat2db/community/query/excel/domain/api/model/QueryDataset.java
package ai.chat2db.community.query.excel.domain.api.model;

import java.util.Date;
import java.util.List;

import lombok.Data;

@Data
public class QueryDataset {

    private Long id;

    private Long workspaceId;

    private String name;

    private String description;

    private Long datasourceId;

    private String databaseName;

    private String schemaName;

    private String tableName;

    private String sourceObjectType;

    private String status;

    private Integer version;

    private String sourceSchemaHash;

    private List<QueryDatasetField> fields;

    private List<DatasetFilter> baseFilters;

    private Long ownerId;

    private Date gmtCreate;

    private Date gmtModified;
}

File: chat2db-community-client/src/typings/queryDataset.ts
export interface QueryDatasetField {
  fieldId?: string;
  sourceColumn?: string;
  displayName?: string;
  dataType?: string;
  role?: string;
  aggregation?: string;
  filterable?: boolean;
  sortable?: boolean;
  visible?: boolean;
  numberFormat?: string;
  nullDisplay?: string;
}

export interface DatasetFilter {
  fieldId?: string;
  operator?: string;
  value?: string;
  values?: string[];
}

export interface QueryDataset {
  id?: number;
  workspaceId?: number;
  name?: string;
  description?: string;
  datasourceId?: number;
  databaseName?: string;
  schemaName?: string;
  tableName?: string;
  sourceObjectType?: string;
  status?: string;
  version?: number;
  sourceSchemaHash?: string;
  fields?: QueryDatasetField[];
  baseFilters?: DatasetFilter[];
  ownerId?: number;
  gmtCreate?: string;
  gmtModified?: string;
}

export interface QueryDatasetListParams {
  workspaceId?: number;
  searchKey?: string;
  pageNo: number;
  pageSize: number;
}

export interface PreviewResult {
  rows: Record<string, unknown>[];
  total: number;
  pageNo: number;
  pageSize: number;
  columns: string[];
}
